x/cronos/keeper/precompiles/relayer.go

```diff
@@ -60,18 +62,31 @@ func (bc *RelayerContract) Address() common.Address {
 }
 
 // RequiredGas calculates the contract gas use
-func (bc *RelayerContract) RequiredGas(input []byte) uint64 {
-	// base cost to prevent large input size
-	baseCost := uint64(len(input)) * bc.kvGasConfig.WriteCostPerByte
+func (bc *RelayerContract) RequiredGas(input []byte) (gas uint64) {
 	if len(input) < prefixSize4Bytes {
 		return 0
 	}
+	intrinsicGas, err := core.IntrinsicGas(input, nil, false, true, true)
```

Comment:
I'm not sure what is the logic there to compute the intrinsic gas but the gas cost should be related to the cosmos cost (ie storage ops), not the ethereum one?

Reply:
patch to be closed how [eth](https://github.com/ethereum/go-ethereum/blob/v1.10.26/core/state_transition.go#L118) calculate gas, while [cosmos](https://github.com/cosmos/cosmos-sdk/blob/v0.47.5/x/auth/ante/basic.go#L96) only use len(input) * 10, you could check Details [here](https://github.com/crypto-org-chain/cronos/pull/1209#issue-1944175530)

Reply:
Got it.
I think the question is more if ETH gas calculation is relevant here or if we should only use cosmos computation method.

Reply:
we will have bigger gap since intrinsic gas is charged when sign evm tnx, while cosmos doesn't have different price for zero and non-zero.

Reply:
I still don't understand the logic of returning `total - intrinsicGas`

Per definition `The intrinsic gas for a transaction is the amount of gas that the transaction uses before any code run`

In that case should we replace the baseCost value by the intrinsic gas value?

Reply:
intrinsic gas will be charged, it's to minimize the diff between cosmos and eth?

Reply:
I see,

In that case, in that case, does it make sense to keep the intrinsic gas as "extra charge" since evm operations is more resource consuming that cosmos native operation?

Reply:
we call cosmos msg server in precompile, and won't we keep gas close to cosmos cost? intrinsic has bigger variant based on ratio of zero and non-zero, can you explain your formula?

Reply:
Precompile is a cosmos msg executed in an evm environment right?

In that case it should be most costly than a standard cosmos msg which do not need evm initialisation. 

Basically per definition every evm msg needs to pay the intrinsic gas cost. since calling precompile is an evm msg, I don't know why we remove this cost

Reply:
But I agree that since we already pay the intrinsic cost per default, we should probably remove the baseCost w

Reply:
But I agree that since we already pay the intrinsic cost per default, we should probably remove the baseCost which may be redundant. Any feedback @yihuang ?

Reply:
LGTM based on the resulting approximation actually.